impeller/entity/shaders/geometry/polyline.comp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+layout(local_size_x = 512, local_size_y = 1) in;
+layout(std430) buffer;
+
+#include <impeller/path.glsl>
+#include <impeller/prefix_sum.glsl>
+
+#define MEMORY_SIZE 1024
+
+layout(binding = 0) readonly buffer Quads {
+  uint count;
+  QuadData data[];
+}
+quads;
+
+layout(binding = 1) readonly buffer Lines {
+  uint count;
+  LineData data[];
+}
+lines;
+
+layout(binding = 2) readonly buffer Components {
+  uint count;
+  PathComponent data[];
+}
+components;
+
+struct PolylineComponent {
+  vec2 pt;
+  // This is the end of a countour.
+  //
+  // that is path.close was called with this as the final
+  // point.
+  bool contour_end;
+
+  // This point is the end of a line segment.
+  bool segment_end;
```

Comment:
We need to track both pieces of this to determine where to insert stroke joins and end caps.

I think that means we also need to track segment start too? I could re-arrange this to be a bit field instead if we need to pack more in.